src/obj.rs

```diff
@@ -119,21 +119,22 @@ impl BtfObj {
 
             if bt.name_off > 0 {
                 let name_off = bt.name_off;
-                let name = str_cache.get(&name_off);
-                if let Some(name) = name {
-                    match strings.get_mut(name) {
+                // Look for the name in our own cache, and if not found try
+                // looking into the base one (if any).
+                let name = str_cache
+                    .get(&name_off)
+                    .or_else(|| base.as_ref().and_then(|base| base.str_cache.get(&name_off)));
+
+                match name {
+                    Some(name) => match strings.get_mut(name) {
                         Some(entry) => entry.push(id),
                         None => _ = strings.insert(name.clone(), vec![id]),
```

Comment:
I don't understand the rationale behind this change, sorry.

> 
> Otherwise if a name is defined in the base BTF and a type refers to it
> in a split BTF, the name to id mapping won't be made and
> resolve_types_by_name will fail even though the type and the name are
> defined.

If a name is defined in the base BTF, resolve_types_by_name() will look for it:

``` rust
    /// Find a list of BTF types using their name as a key.
    pub fn resolve_types_by_name(&self, name: &str) -> Result<Vec<Type>> {
        let mut types = Vec::new();

        if let Some(base) = &self.base {
            if let Ok(mut types_base) = base.resolve_types_by_name(name) {
                types.append(&mut types_base);
            }
        }
        if let Ok(mut types_obj) = self.obj.resolve_types_by_name(name) {
            types.append(&mut types_obj);
        }
 [...]
}

```

This would unnecessarily increase the size of all split btf files with (potentially a lot of) symbol names that are already present in the base.

Reply:
Thanks for the explanation, and sorry for the delay

Reply:
Thanks! Opened a dedicated PR in #24.